src/pe/resource.rs

```diff
@@ -787,7 +787,11 @@ impl<'a> ResourceString<'a> {
             },
             4,
         );
-        let value = bytes.pread_with::<&[u8]>(*offset, real_value_len as usize)?;
+
+        // Some resource compilers (like mono) don't pad the last value to 4-byte alignment.
+        // If there aren't enough bytes for the aligned length, use what's available.
+        let actual_read_len = real_value_len.min(bytes.len() - *offset);
+        let value = bytes.pread_with::<&[u8]>(*offset, actual_read_len)?;
```

Comment:
Instead of assuming that the string actually ends at the slice bound, I'd rather modifying padding directly.

You can avoid adding padding bytes if there are no spaces for it.

https://github.com/m4b/goblin/blob/589ffe5787d1d9ab2007a7f98051fa60d073938a/src/pe/resource.rs#L782-L789